purchase_request_procurement/models/procurement_order.py

```diff
@@ -27,35 +27,50 @@ def _prepare_purchase_request_line(self):
             'date_required': self.date_planned,
             'product_uom_id': product.uom_po_id.id,
             'product_qty': procurement_uom_po_qty,
-            'request_id': self.request_id.id,
+            'request_id': pr.id,
             'procurement_id': self.id
         }
 
     @api.multi
     def _prepare_purchase_request(self):
         self.ensure_one()
-
+        gpo = self.rule_id.group_propagation_option
+        group_id = (gpo == 'fixed' and self.group_id.id) or \
+                   (gpo == 'propagate' and self.group_id.id) or False
         return {
             'origin': self.origin,
             'company_id': self.company_id.id,
             'picking_type_id': self.rule_id.picking_type_id.id,
+            'group_id': group_id,
         }
 
-    @api.multi
-    def _search_existing_purchase_request(self):
+    @api.model
+    def _make_pr_get_domain(self, values):
         """
         This method is to be implemented by other modules that can
         provide a criteria to select the appropriate purchase request to be
         extended.
         :return: False
         """
-        return False
+        domain = (
+            ('state', '=', 'draft'),
+            ('picking_type_id', '=', values['picking_type_id']),
+            ('company_id', '=', values['company_id']),
+        )
+        gpo = self.rule_id.group_propagation_option
+        group_id = (gpo == 'fixed' and self.group_id.id) or \
```

Comment:
If the method is @api.model why do you use self.foo?? I think that is not correct